### PR TITLE
Add TypedResults.Extensions

### LINKED ISSUE #47050: Add Extensions to Http.TypedResults
## Background and Motivation
`TypedResults` has a very similar api as `Results`, but it lacks an `Extensions` property.

This property would be useful for consistency, all endpoints can use `TypedResults` even when using customized results via an extension. Besides that there seem to be plans the recommend and add a codefixer for using `TypedResults` where possible. (#45217)

## Proposed API

```diff
namespace Microsoft.AspNetCore.Http;

public static class TypedResults
{
+    public static IResultExtensions Extensions { get; } = new ResultExtensions();
}
```

## Usage Examples

```csharp
app.MapGet("/",  () => TypedResults.Extensions.CustomResult());
```

## Alternative Designs

Alternatively a new specific interface could be used instead:

```diff
namespace Microsoft.AspNetCore.Http;

+ public interface ITypedResultExtensions { }

public static class TypedResults
{
+    public static ITypedResultExtensions Extensions { get; } = new ResultExtensions();
}
```

## Risks

When going for the proposed solution, existing extensions automatically become available, even if they are not 'Typed'.

### PR DESCRIPTION
**Pull request by dnperfors**
Make TypedResults more consistent with Results by adding extensions support. Extensions on Results.Extensions will also work on TypedResults.

Fixes #47050

### CODE REVIEW

File: src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs [hunk 1]
```diff
@@ -3743,6 +3743,31 @@ public async Task CanExecuteRequestDelegateWithResultsExtension()
         Assert.Equal(@"""Hello Tester. This is from an extension method.""", decodedResponseBody);
     }
 
+    [Fact]
+    public async Task CanExecuteRequestDelegateWithTypedResultsExtension()
```
Comment (captainsafia) [inline src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs:3747]: Can you move this test over to this file: https://github.com/dotnet/aspnetcore/blob/f876ff14f8d4d28b410f294f2ec92b3aaf619060/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTests.Responses.cs?

Context: we now have a compile-time source generator for minimal APIs and placing your test in the file above will validate we have the same behavior between the compile-time and runtime implementations.

Comment (dnperfors) [reply src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs:3747]: Could you give me some guidance on how to provide an extra class to the 'source' of my test? 
Basically I need to use the following source:
```
string source = """app.MapGet("/", () => TypedResults.Extensions.TestResult();""";
```
But I don't know where I can add the `TestResult` extension method.

Comment (captainsafia) [reply src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs:3747]: You can add the definition here: https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTestBase.cs#L253

That's the mock source file the `MapGet` code gets written into.

Comment (dnperfors) [reply src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs]: I was just checking if there was not other obvious place I missed :)

**Review by captainsafia — APPROVED** [review]
Thank you for your contribution! 🙏🏽

**Pull request merged**